### PR TITLE
Fix a11y for Desktop Menu

### PR DESCRIPTION
**Pull request by mparvazi**
### Keyboard navigation:
- **Main menu links [depth=0]**
 1- Change focus by TAB
 2- Open menu with ARROW_DOWN (focus on first link [depth=1]) and ARROW_UP (focus on last link [depth=1])
- **Inside opened menu**
 1- Change focus by TAB
 2- Close menu with ESC
 3- Open dropdown with ARROW_RIGHT and ARROW_LEFT

https://user-images.githubusercontent.com/85633460/167289613-2f07b536-4ad6-41c5-8f80-44ae0603d652.mp4

### CODE REVIEW

**Comment by mparvazi** [comment]
@NeOMakinG 

Give me a screenshot for focus elements on Safari.
Need to compare with Firefox and Chrome.
Should define outline style for hummingbird to make it similar on all browsers. 

Hopefully Firefox in new version made outline style as like as Chrome.

File: modules/ps_mainmenu/ps_mainmenu.tpl [hunk 1]
```diff
@@ -1,104 +1,140 @@
...
   </button>
 </div>
 
-<div class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas" tabindex="-1" id="mobileMenu"
-  aria-labelledby="mobileMenuLabel">
+<div
+  class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas" 
+  tabindex="-1" 
+  id="mobileMenu"
+  aria-labelledby="mobileMenuLabel"
```
Comment (NeOMakinG) [inline modules/ps_mainmenu/ps_mainmenu.tpl:126]: I guess no content is linked to that label, we should either use aria-label or use a good element to describe this canvas, wdyt?

File: src/js/modules/ps_mainmenu.ts [hunk 2]
```diff
@@ -1,45 +1,91 @@
...
+  /**
+   * Handle Keyboard events for main menu links
+   * Find all links with [data-depth: 0]
+   */
+  const mainMenuLinks = document.querySelectorAll(desktopMenuMap.dropdownItemAnchor(0));
+  mainMenuLinks.forEach((mainMenuLinkElement: HTMLElement) => {
+    const menuContainerElement = mainMenuLinkElement.nextElementSibling;
+
+    /* Open the menu container when Arrow Up or Arrow Down key is pressed */
+    mainMenuLinkElement.addEventListener('keydown', (event: KeyboardEvent) => {
```
Comment (NeOMakinG) [inline src/js/modules/ps_mainmenu.ts:62]: Do we really need this? Bootstrap dropdowns should naturally handle keyboard navigation

Comment (mparvazi) [reply src/js/modules/ps_mainmenu.ts:62]: The current structure is not based on BS standard for dropdowns,
If we add 'dropdown' selector to main links parent container to enable BS keyboard navigation,
then we will have problem for controlling the menu container, so need to do a refacto and change current structure to BS navbar.

File: modules/ps_mainmenu/ps_mainmenu.tpl (hunk 1 above)
Comment (mparvazi) [reply modules/ps_mainmenu/ps_mainmenu.tpl:126]: Yep!
It's just a container, I think something like this is enough:
`<div id="mobileMenu" class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas">`

Comment (NeOMakinG) [reply modules/ps_mainmenu/ps_mainmenu.tpl:126]: Probably, as the id will be taken in account when screen readers talks?

File: src/js/modules/ps_mainmenu.ts (hunk 2 above)
Comment (NeOMakinG) [reply src/js/modules/ps_mainmenu.ts:62]: Do you see some constraints on moving to the BS structure? If we can use the exact same style but with a cleaner bootstrap structure, that could be a great maintainable solution

I would probably accept your PR as a first workaround, but if we can do something more cleaner, sticking to BS, maybe this is the right choice

wdyt @Hlavtox ?

**Review by NeOMakinG — APPROVED** [review]
Not blocking, it's a great working thing with the current menu implementation, well-done.

But I'm wondering if we didn't do a mistake by not using the bootstrap navbar markup directly, and trying to have the same mockup styling, maybe this would be a great improvement in the future

File: src/js/modules/ps_mainmenu.ts (hunk 2 above)
Comment (mparvazi) [reply src/js/modules/ps_mainmenu.ts:62]: The current structure is fine,
If you try to use BS Navbar with dropdowns in main links, then you can not have menu-container with rows and columns.

In this PR I tried to simulate BS dropdown behavior for dropdowns in main-links.

File: modules/ps_mainmenu/ps_mainmenu.tpl (hunk 1 above)
Comment (mparvazi) [reply modules/ps_mainmenu/ps_mainmenu.tpl:126]: `<div class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas" id="mobileMenu">`

Comment (mparvazi) [reply modules/ps_mainmenu/ps_mainmenu.tpl:126]: I don't have any idea about this one!

Comment (NeOMakinG) [reply modules/ps_mainmenu/ps_mainmenu.tpl:126]: I guess yes

**Comment by mparvazi** [comment]
@Hlavtox 
Wdyt?
Any changes needed?

**Pull request merged**